include/aspect/initial_conditions/interface.h

```diff
@@ -93,6 +94,31 @@ namespace aspect
     };
 
 
+    /**
+     * Create a class inheriting from dealii.Function to translate the
+     * interface call to dealii.Function format
+     */
+    template <int dim>
+    class InterfaceFunction : public dealii::Function<dim, double>
+    {
+      public:
+        InterfaceFunction (const Interface<dim> &interface,
+                           const unsigned int selected_component,
+                           const unsigned int n_components);
```

Comment:
I mentioned above specifically that that does not work. Remember that Interface<dim> does not implement initial_temperature, expecting the inheriting classes to do so. As such, it is necessary that whatever logic is used to reference the initial_temperature method must be able to handle calling the correct class's method. Using the newly implemented class, this is automatic, but I could not find a way to  use std_cxx11::bind to do so.